packages/browser-wallet/src/popup/store/identity.ts

```diff
@@ -42,3 +49,11 @@ export const identityNamesAtom = selectAtom(identitiesAtom, (identities) => {
 });
 
 export const isRecoveringAtom = atomWithChromeStorage<boolean>(ChromeStorageKey.IsRecovering, false, true);
+export const recoveryStatusAtom = atomWithChromeStorage<RecoveryStatus | undefined>(
+    ChromeStorageKey.RecoveryStatus,
+    undefined,
+    true
+);
+export const setRecoveryPayloadAtom = atom<null, RecoveryPayload, Promise<void>>(null, (_, set, payload) =>
+    set(recoveryStatusAtom, { payload })
```

Comment:
Does this only update the payload part of the status?

Reply:
No, it resets everything else.